### DISCUSSION TITLE
Styling the expander. How?

### DISCUSSION THREAD

**Discussion by timahrentlov** [AvaloniaUI/Avalonia#11700] [Q&A] [answered]
I need some help

1. How would I go about removing the border around the expander header and it's chevron?
2. Where should I look in the source code (themes?) and how should I interpret the code (translate to styling path) in order to figure these things out for my-self ?
3. Is there a book I can read, a video I can watch or anything to get me up to speed?

Thanks

**Comment by robloo** [comment, marked as answer]
If you are using the Fluent theme (highly recommended) the source for the Expander "style" now called control theme in Avalonia is here:

https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/Expander.xaml

This control was redesigned a bit for 11.0 to match closer to WinUI upstream. It is divided into two main areas: the header and the content area. The header is essentially just a specially themed `ToggleButton` defined by the `FluentExpanderToggleButtonTheme` -- which you could customize. However, we use a trick from WinUI here too and have what are called "lightweight styling resources".

These "lightweight styling resources" allow you to override a resource by key in the visual tree to change the look of a control without having to re-template or heavily change the style. In the case of the header border take a look at these resources which you can override:

 * `ExpanderHeaderBorderBrush` https://github.com/AvaloniaUI/Avalonia/blob/dc9b47b9bef586b404e04372bbd4bb47cd14f3fb/src/Avalonia.Themes.Fluent/Controls/Expander.xaml#L55
 * `ExpanderHeaderBorderThickness` https://github.com/AvaloniaUI/Avalonia/blob/dc9b47b9bef586b404e04372bbd4bb47cd14f3fb/src/Avalonia.Themes.Fluent/Accents/FluentControlResources.xaml#L395-L398

Note that some resources (brushes) need to change based on light/dark theme variant and these are separated out into the FluentControlResources file.

There are also resources that can control the chevron if you take a look in the code I pointed you to above for the header border.

**Reply by timahrentlov** [reply]
That is a very instructive answer. I will dig into it right away. Thank you for your time, @robloo !

**Comment by timahrentlov** [comment]
Update. 

I also tried to modify the angle of the chevron. I wanted it to go from 90 to 0 and back, instead of doing 180's animations. However, if I aim with    

<Style Selector="Expander ToggleButton:checked /template/ Path#ExpandCollapseChevron"> 

to modify the animations, it will execute both the old animations and the new ones I supplied with the style. In effect, the animation styles are not overridden but added to the styles. So I'm currently investigating how to clear the old animations. 

Unfortunately the angels are not provided via resources that would be easy to override. Anyway, if I don't find a way to clear animations, I will simply rewrite the control or make all new control templates.

**Reply by robloo** [reply]
Yea, you are getting a bit advanced in the customizations. There is no issue in this case to just make your own control theme and a new control template. That gives you the full power to do as you want.

**Reply by timahrentlov** [reply]
Yup. It tought me a thing or two. No regrets. Full speed ahead :)
Thanks @robloo.
